### PR TITLE
Popup shader creation dialog at selecting a shader resource at the Create Resource Dialog in FileSystem

### PR DESCRIPTION
**Pull request by Chaosus**
~Added a new option to that menu to easy create a shader using a new shader creation dialog~:
filesystemdock_shader2

### CODE REVIEW

**Comment by KoBeWi** [comment]
Wouldn't it be possible to instead open the shader dialog when user selects New Resource and picks Shader? Not sure if creating a Shader is common enough to have a dedicated entry.

**Comment by Chaosus** [comment]
If the user selects a new resource the dialog with two options - Shader and VisualShader is appearing - but I think it's non-intuitive to popup a shader dialog after the click on either of them since the resources would be created immediately for any other resource type in that dialog.

**Comment by KoBeWi** [comment]
When you click the New Resource option, you first select the type and then file dialog opens where you select path. For shaders we could just display the shader dialog instead of file dialog.

**Comment by Chaosus** [comment]
@KoBeWi Done, as you said

File: editor/filesystem_dock.h [hunk 1]
```diff
@@ -52,6 +52,7 @@
 #include "editor_dir_dialog.h"
 #include "editor_file_system.h"
 #include "script_create_dialog.h"
+#include "shader_create_dialog.h"
```
Comment (KoBeWi) [inline editor/filesystem_dock.h:55]: The include should actually be in the `.cpp` file to avoid compilation dependency (other includes are wrong too).

Comment (Chaosus) [reply editor/filesystem_dock.h]: Fixed

Comment (Chaosus) [reply editor/filesystem_dock.h]: For others may be better to open a specified PR?

Comment (KoBeWi) [reply editor/filesystem_dock.h]: Yeah, I noticed that lots of includes need a cleanup, so it could be left for another PR.

**Review by KoBeWi — APPROVED** [review]

**Pull request merged**